autometa/common/external/diamond.py

```diff
@@ -28,204 +28,359 @@
-    verbose : boolean
-        log progress to terminal `verbose` (the default is False).
+        Path to output file
+        '</path/to/output/file>'
+    blast_type : str, optional
+        blastp to align protein query sequences against a protein reference database,
+        blastx to align translated DNA query sequences against a protein reference database, by default 'blastp'
+    evalue : float, optional
+        cutoff e-value to count hit as significant, by default float('1e-5')
+    maxtargetseqs : int, optional
+        max number of target sequences to retrieve per query by diamond, by default 200
+    cpus : int, optional
+        Number of processors to be used, by default uses all the processors of the system
+    tmpdir : str, optional
+        Path to temporary directory
+        '</path/to/temporary/directory>' by default tempfile.mkdtemp()
+    force : bool, optional
+        overwrite existing diamond results, by default False
+    verbose : bool, optional
+        log progress to terminal, by default False
 
     Returns
     -------
     str
-        `outfpath`
+        Path to BLAST results
 
     Raises
-    -------
+    ------
     FileNotFoundError
         `fasta` file does not exist
     ValueError
         provided `blast_type` is not 'blastp' or 'blastx'
-    OSError
-        Diamond execution failed
+    subprocess.CalledProcessError
+        Failed to run blast
     """
     if not os.path.exists(fasta):
         raise FileNotFoundError(fasta)
     if os.path.exists(outfpath) and not force:
-        empty = not os.stat(outfpath).st_size
-        if not empty:
+        if not os.path.getsize(outfpath):
             if verbose:
-                logger.warning(f'FileExistsError: {outfpath}. To overwrite use --force')
+                logger.warning(f"FileExistsError: {outfpath}. To overwrite use --force")
             return outfpath
     blast_type = blast_type.lower()
-    if blast_type not in ['blastp','blastx']:
-        raise ValueError(f'blast_type must be blastp or blastx. Given: {blast_type}')
+    if blast_type not in ["blastp", "blastx"]:
+        raise ValueError(f"blast_type must be blastp or blastx. Given: {blast_type}")
     if verbose:
-        logger.debug(f'Diamond{blast_type.title()} {fasta} against {database}')
+        logger.debug(f"Diamond{blast_type} {fasta} against {database}")
     cmd = [
-        'diamond',
-        'blastp',
-        '--query',
+        "diamond",
+        blast_type,
+        "--query",
         fasta,
-        '--db',
+        "--db",
         database,
-        '--evalue',
+        "--evalue",
         evalue,
-        '--max-target-seqs',
+        "--max-target-seqs",
         maxtargetseqs,
-        '--threads',
+        "--threads",
         cpus,
-        '--outfmt',
-        '6',
-        '--out',
+        "--outfmt",
+        "6",
+        "--out",
         outfpath,
-        '--tmpdir',
-        tmpdir]
+        "--tmpdir",
+        tmpdir,
+    ]
```

Comment:
It seems to me, the only time you will need to add `--tmpdir tmpdir` to the command is when this is supplied by the user. Otherwise, I think diamond will handle this. Similarly, you could place the logic _within_ this function if you would like to set the tmpdir using the `tempfile.mkdtemp` method. The default `tmpdir` parameter could be `None`. If the default were to be `None`, then you could assign the `tmpdir` variable to `tempfile.mkdtemp()`.